packages/clay-core/docs/treeview.mdx

```diff
@@ -0,0 +1,378 @@
+							shape="user-icon"
+							size="sm"
+						>
+							UT
+						</Sticker>
+						ultrices dui sapien
+					</TreeView.Item>
+				</TreeView.Group>
+			</TreeView.Item>
+			<TreeView.Item key="emily-young">
+				<Sticker displayType="primary" shape="user-icon" size="sm">
+					MC
+				</Sticker>
+				maecenas pharetra convallis
+			</TreeView.Item>
+		</TreeView.Group>
+	</TreeView.Item>
+</TreeView>
+```
+
+### Dynamic
+
+The TreeView and Group components when used with dynamic items are populated from hierarchical data. The component consumes the data using the `items` property.
+
+For the component to be dynamic it is necessary to convert `children` to Render prop, which must be a function that will receive the `item` of the current iterator.
+
+```jsx
+function Example() {
+	const items = [
+		{
+			children: [
+				{
+					children: [{name: 'ultrices dui sapien'}],
+					name: 'fusce ut placerat',
+				},
+				{name: 'maecenas pharetra convallis'},
+			],
+			name: 'nisi quis eleifend',
+		},
+	];
+
+	return (
+		<TreeView items={items} nestedKey="children">
+			{(item) => (
+				<TreeView.Item key={item.name}>
+					{item.name}
+
+					{item.children && (
+						<TreeView.Group items={item.children}>
+							{(item) => (
+								<TreeView.Item key={item.name}>
+									{item.name}
+								</TreeView.Item>
+							)}
+						</TreeView.Group>
+					)}
+				</TreeView.Item>
+			)}
+		</TreeView>
+	);
+}
+```
+
+#### Asynchronous Item
+
+Asynchronous Item loading is quite advantageous to decrease the initial payload and decrease memory space, this is more efficient when a tree is very large.
+
+The TreeView doesn't know when an Item is asynchronous so a deduction is made that the developer must deal with on the implementation side to tell whether the Item is asynchronous or not. The `onLoadMore` property is called every time the Item is a leaf of the tree and depending on the method's return it will make some assumptions.
+
+-   Returning `void`, `null` or `undefined` the TreeView will do nothing.
+-   Returning the `item` will add to the tree.
+
+Adding a new asynchronous Item to the tree the `onItemsChange` method is respectively called to update the tree with a new value if the `items` is controlled.
+
+<div class="clay-site-alert alert alert-info">
+	<strong class="lead">Warning</strong>
+	If you have an error in the asynchronous call of the <code className="gatsby-code-text">
+		onLoadMore
+	</code> method, only the suppression is done and an error is thrown on the console.
+</div>
+
+```jsx
+<TreeView
+	onLoadMore={async (item) => {
+		return await fetch(`example.com/tree/item?parent_id=${item.id}`);
+	}}
+>
+	{...}
+</TreeView>
+```
+
+### Item
+
+Item is very flexible and can behave in different ways to simplify usage for different use cases depending on the data.
+
+Item also helps to define the unique `key` of the component across the tree which is also used to tell which items are expanded and selected.
```

Comment:
```diff
- expanded and selected
+ expanded or selected
```
In that case, I think we should leave `and` because it's used for both and not one or the other, does it make sense? I may be getting it wrong.